specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/job.json

```diff
@@ -0,0 +1,931 @@
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/JobOperationsErrorResponse"
+                        }
+                    },
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/JobSubmissionJsonResponse"
+                        }
+                    }
+                }
+            }
+        },
+        "/ws/v1/cluster/apps/{appId}/state": {
+            "get": {
+                "tags": [
+                    "Job"
+                ],
+                "operationId": "Job_GetAppState",
+                "description": "Gets application state from the specified HDInsight cluster.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "x-ms-examples": {
+                    "Gets application state from the specified HDInsight cluster.": {
+                        "$ref": "./examples/HDI_Job_GetAppState.json"
+                    }
+                },
+                "parameters": [
+                    {
+                        "name": "appId",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The id of the job."
+                    }
+                ],
+                "responses": {
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/JobOperationsErrorResponse"
+                        }
+                    },
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/AppState"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "AppState": {
+            "description": "The State of the application.",
+            "properties": {
+                "state": {
+                    "type": "string",
+                    "description": "The State of the application. Possible values are “NEW”, “NEW_SAVING”, “SUBMITTED”, “ACCEPTED”, “RUNNING”, “FINISHED”, “FAILED”, “KILLED”"
+                }
+            }
+        },
+        "JobOperationsErrorResponse": {
+            "description": "Describes the format of Error response.",
+            "type": "object",
+            "properties": {
+                "error": {
+                    "description": "Error message indicating why the operation failed.",
+                    "type": "string"

```

Comment:
What is the set of possible error messages? Is this a human-readable string or is the expectation that an application should be able to act based on the value of *error*?

Reply:
They are human-readable strings. I am not sure about possible values.

---
In reply to: [237731777](https://github.com/Azure/azure-rest-api-specs/pull/4489#discussion_r237731777) [](ancestors = 237731777)